benchcab/workdir.py

```diff
@@ -28,60 +29,59 @@ def setup_src_dir(root_dir=internal.CWD):
-):
-    """Generate the directory structure used of `benchcab`."""
+def fluxsite_directory_tree_list(fluxsite_tasks: list[Task], root_dir=internal.CWD) -> set:
+    """Generate a list of all the work directories used for fluxsite tests"""
 
-    run_dir = Path(root_dir, internal.RUN_DIR)
-    if not run_dir.exists():
-        os.makedirs(run_dir)
-
-    fluxsite_run_dir = Path(root_dir, internal.FLUXSITE_RUN_DIR)
-    if not fluxsite_run_dir.exists():
-        os.makedirs(fluxsite_run_dir)
-
-    fluxsite_log_dir = Path(root_dir, internal.FLUXSITE_LOG_DIR)
-    if not fluxsite_log_dir.exists():
-        print(
-            f"Creating {fluxsite_log_dir.relative_to(root_dir)} directory: {fluxsite_log_dir}"
-        )
-        os.makedirs(fluxsite_log_dir)
-
-    fluxsite_output_dir = Path(root_dir, internal.FLUXSITE_OUTPUT_DIR)
-    if not fluxsite_output_dir.exists():
-        print(
-            f"Creating {fluxsite_output_dir.relative_to(root_dir)} directory: {fluxsite_output_dir}"
-        )
-        os.makedirs(fluxsite_output_dir)
-
-    fluxsite_tasks_dir = Path(root_dir, internal.FLUXSITE_TASKS_DIR)
-    if not fluxsite_tasks_dir.exists():
-        print(
-            f"Creating {fluxsite_tasks_dir.relative_to(root_dir)} directory: {fluxsite_tasks_dir}"
-        )
-        os.makedirs(fluxsite_tasks_dir)
-
-    fluxsite_analysis_dir = Path(root_dir, internal.FLUXSITE_ANALYSIS_DIR)
-    if not fluxsite_analysis_dir.exists():
-        print(
-            f"Creating {fluxsite_analysis_dir.relative_to(root_dir)} directory: "
-            f"{fluxsite_analysis_dir}"
-        )
-        os.makedirs(fluxsite_analysis_dir)
-
-    fluxsite_bitwise_cmp_dir = Path(root_dir, internal.FLUXSITE_BITWISE_CMP_DIR)
-    if not fluxsite_bitwise_cmp_dir.exists():
-        print(
-            f"Creating {fluxsite_bitwise_cmp_dir.relative_to(root_dir)} directory: "
-            f"{fluxsite_bitwise_cmp_dir}"
-        )
-        os.makedirs(fluxsite_bitwise_cmp_dir)
-
-    print("Creating task directories...")
+    # Create the list of directories as sets because the order is not important
+    # and we want to avoid duplications.
+    fluxsite_paths = []
+
+    # Run directory
+    fluxsite_paths.append(Path(root_dir, internal.RUN_DIR))
+
+    # Fluxsite run directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_RUN_DIR))
+
+    # Fluxsite log directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_LOG_DIR))
+
+    # Fluxsite output directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_OUTPUT_DIR))
+
+    # Fluxsite tasks directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_TASKS_DIR))
+
+    # Fluxsite analysis directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_ANALYSIS_DIR))
+
+    # Fluxsite bit-wise comparison directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_BITWISE_CMP_DIR))
+
+    # Fluxsite tasks directories. append all of them as a set().
```

Comment:
After writing this code again for the spatial tasks, I have a feeling that creating the of task directories should instead be in the setup function in fluxsite.py:
https://github.com/CABLE-LSM/benchcab/blob/04dabaf556041fb905c1f3772f01ae8d34e01b79/benchcab/fluxsite.py#L128

This would probably remove the need to check for nested lists in the path list? This section of code is largely specific to fluxsite tasks and so it makes sense to put it in the fluxsite module.